api/instance/v1/server_utils.go

```diff
@@ -51,3 +52,51 @@ func (s *API) WaitForServer(req *WaitForServerRequest) (*Server, error) {
 
 	return server.(*Server), err
 }
+
+// DoActionAndWaitRequest is used by DoActionAndWait method
+type DoActionAndWaitRequest struct {
+	ServerID string
+	Zone     utils.Zone
+	Action   ServerAction
+	Timeout  time.Duration
+}
+
+// DoActionAndWait start an action and wait for the server to be in the correct "terminal state"
+// expected by this action.
+func (s *API) DoActionAndWait(req *DoActionAndWaitRequest) error {
```

Comment:
If underlying method is `ServerAction` should this be `ServerActionAndWait` ?
or maybe rename underlying action